### ISSUE TITLE
Backend Progress

### ISSUE THREAD

**Issue by emmaxshen** [e-shen2022/emma_blog#48]
***Read comments for line by line explanations*** 

1. Import Packages
<img width="361" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/7b1755ba-6581-45d3-a593-0b3fbf3f2e33">

- Numpy to save image data in arrays 
- Tensorflow and keras to create and train model 

2. Train data
<img width="568" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/0ecfa74f-fb6a-43ac-b4f8-708181aed3ed">

- Resize image input to fit dimensions for NN training
- Specify number of images per batch, and how many batches model will train 
- Converts images to grayscale

3. Creating the CNN model 
<img width="723" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/433c5f95-7883-469c-b9a5-c1b510ba4ab6">

- Sequential model where layers increase in complexity of what pattern can recognize
- Adds convolutional layers to detect local patterns
- Add pooling layers to downsample data to look at small areas of image 
- Dropout layer to regularize data and prevent overfitting
- Flatten layer to reshape to 1D array to prepare for connecting all layers 
- Dense layer to connect all layers, last layer specifies 7 neurons which is 7 different possible emotion outputs

4. Compiling the model for training 
<img width="612" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/6452989f-0be9-4720-be92-353ae34d1fb2">

- Prepare model for training by defining how it will measure loss, update its weights, and evaluate its prediction performance
- Save learned parameters in model.h5 for graphical user interface
